src/OrleansRuntime/Placement/PlacementDirectorsManager.cs

```diff
@@ -106,19 +106,27 @@ private IActivationSelector ResolveSelector(PlacementStrategy strategy, bool add
 
         private void ResolveBuiltInStrategies()
         {
-            var strategies = new PlacementStrategy[]
+            var statelessWorker = new StatelessWorkerPlacement();
+
+            var placementStrategies = new PlacementStrategy[]
             {
                 RandomPlacement.Singleton,
                 ActivationCountBasedPlacement.Singleton,
-                new StatelessWorkerPlacement(),
+                statelessWorker,
                 PreferLocalPlacement.Singleton
             };
-            foreach (var strategy in strategies)
+
+            foreach (var strategy in placementStrategies)
+                this.ResolveDirector(strategy);
+            
+            var selectorStrategies = new PlacementStrategy[]
             {
-                var director = this.ResolveDirector(strategy);
-	            if (director is IActivationSelector)
-		            this.ResolveSelector(strategy);
-            }
```

Comment:
what was wrong with this approach? it was a nice one to avoid issues if more built in strategies support selection in the future. The only thing that was missing was passing `true` to `ResolveSelector`, right?
Anyway, I think it's OK regardless, even if my preference is the previous approach.